File: Data Structures & Algorithms/majority-element/submission-0.java
class Solution {
    public int majorityElement(int[] nums) {
        HashMap<Integer,Integer>map=new HashMap<>();
        for(int i:nums){
            map.put(i,map.getOrDefault(i,0)+1);
        }
        int maxCount=0;
        int maxNum=nums[0];
        for(int i :nums){
           int count = map.get(i);
           if(count>maxCount){
            maxCount=count;
            maxNum=i;
           }
        }
        return maxNum;
    }
}

File: Data Structures & Algorithms/jump-game/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    canJump(nums) {
        let rt = nums.length-1
        // let lt = nums.length-1
        for(let i=nums.length-1;i>=0;i--){
            let lt = i
            if(nums[lt] + lt >= rt){
                rt=lt
                lt-=1
            }
        }
        if(rt === 0){
            return true
        }
        return false
    }
}
